packages/astro/src/core/build/index.ts

```diff
@@ -205,6 +205,13 @@ class AstroBuilder {
 		};
 
 		const { internals, ssrOutputChunkNames, contentFileNames } = await viteBuild(opts);
+
+		const hasServerIslands = this.settings.serverIslandNameMap.size > 0;
+		// Error if there are server islands but no adapter provided.
+		if(hasServerIslands && !this.settings.adapter) {
```

Comment:
How is this impacted by changes made by Erika about hybrid? Are you checking at any point if the buildOutput is 'server'?

Reply:
IIUC it could also be a static adapter too, so maybe it's more accurate to check against `this.settings.buildOutput`?

Reply:
What is inaccurate about this check exactly?

Reply:
server islands are only meant to work if your adapter is for ssr. eg. vercel has a static adapter

Reply:
yeah exactly, i think you may not need to check for the adapter at all since it may be enforced in a central place (@Princesseuh do you know?)

Reply:
Is there a scenario where `buildOutput === 'server'` but there is not an adapter? I feel weird about throwing an error that says you don't an adapter without checking that there is an adapter.

Reply:
I guess there's not. Made the change.

Reply:
It gets caught in the `setup` method of `AstroBuilder` line 133 on `next`